torch_xla/csrc/ir.cpp

```diff
@@ -102,14 +102,18 @@ std::string Use::ToString() const {
   return ss.str();
 }
 
-const xla::Shape& Value::xla_shape() const { return node->xla_shape(index); }
-
-const xla::Shape& Value::xla_node_shape() const { return node->xla_shape(); }
+const xla::Shape& Value::xla_shape() const {
+  Node* casted = dynamic_cast<Node*>(node.get());
+  return casted->xla_shape(index);
+}
 
-torch::lazy::hash_t Value::hash() const {
-  return torch::lazy::HashCombine(node->hash(), index);
+const xla::Shape& Value::xla_node_shape() const {
+  Node* casted = dynamic_cast<Node*>(node.get());
```

Comment:
this will likely incur some cost but I expect it to be small. Let's verify that on benchmark